### PR TITLE
Refactor: Icon component

### PR DESCRIPTION
**Pull request by josenriagu**
## Description

- minimize bloat in Icon component, import specified icons only as required

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: ui/design-system-core/src/components/Icon/index.tsx [hunk 1]
```diff
@@ -24,18 +21,9 @@ export interface IconProps {
...
 
-const fillOnlyIcons: IconType[] = [
-  'akasha',
-  'discord',
-  'telegram',
-  'twitter',
-  'widget',
-  'metamask',
-];
-
```
Comment (didd) [inline ui/design-system-core/src/components/Icon/index.tsx:35]: Here is what the fill only icons look like on your PR :

image

Can your revisit how you are setting the solid prop for those and if the style is being applied correctly ?

Comment (josenriagu) [reply ui/design-system-core/src/components/Icon/index.tsx:35]: Alright, let me take a look

File: ui/apps/articles/src/pages/article-card-settings.tsx [hunk 2]
```diff
@@ -43,8 +53,28 @@ const ArticleCardSettingsPage: React.FC<unknown> = () => {
   const { t } = useTranslation('app-articles');
   const { logger } = useRootComponentProps();
 
+  const getLicenseIcon = (type: string) => {
```
Comment (didd) [inline ui/apps/articles/src/pages/article-card-settings.tsx:56]: You can use a map of type to icon element or a switch statement to make it easier to read and cleaner.

File: ui/apps/articles/src/pages/article-card-settings.tsx [hunk 3]
```diff
@@ -43,8 +53,28 @@ const ArticleCardSettingsPage: React.FC<unknown> = () => {
...
+    if (type === 'licenseAllRights') return <LicenseAllRights />;
+    if (type === 'licenseAttribution') return <LicenseAttribution />;
+    if (type === 'licenseNoDerivatives') return <LicenseNoDerivatives />;
+    if (type === 'licenseNoRights') return <LicenseNoRights />;
+    if (type === 'licenseNonCommercial') return <LicenseNonCommercial />;
+    if (type === 'licenseShareAlike') return <LicenseShareAlike />;
+    if (type === 'licenseSomeRights') return <LicenseSomeRights />;
+    if (type === 'LicenseWtfpl') return <LicenseWtfpl />;
```
Comment (didd) [inline ui/apps/articles/src/pages/article-card-settings.tsx]: You can use a map of type to icon element or a switch statement to make it easier to read and cleaner.

File: ui/design-system-components/src/components/Profile/Links/index.tsx [hunk 4]
```diff
@@ -17,24 +23,39 @@ export type LinksProps = {
...
+  const getIconFromLink = (href: string) => {
+    const host = new URL(href).host;
+
+    if ('github.com'.includes(host)) return { icon: <Github />, solid: false };
+    if ('twitter.com'.includes(host)) return { icon: <Twitter />, solid: true };
+    if ('discordapp.com'.includes(host)) return { icon: <Discord />, solid: true };
+    if ('telegram.me'.includes(host) || 't.me'.includes(host))
+      return { icon: <Telegram />, solid: true };
+    return { icon: <LinkIcon />, solid: false };
+  };
```
Comment (didd) [inline ui/design-system-components/src/components/Profile/Links/index.tsx]: This function expression will be re-created needlessly every time the component re-renders. Either we need to put it inside a useCallback which I think isn't worth it or create a function outside of the component.

**Comment by themonster2015** [comment]
Screenshot 2023-11-26 at 01 31 31

Screenshot 2023-11-26 at 01 39 41


I noticed now there's a black outline when displaying the icon inside the toggle in dark and light modes in the settings app :)

**Review by didd — DISMISSED** [review]

**Review by didd — APPROVED** [review]

**Pull request merged**